app/controllers/lessons_controller.rb

```diff
@@ -1,5 +1,14 @@
 class LessonsController < ApplicationController
   def index
-    @categories = Category.all.paginate(page: params[:page], per_page: 10)
+    if params[:filter] == "Learned"
+       @categories = current_user.lessons.where.not(lessons: {result: nil})
```

Comment:
change
`@categories = current_user.lessons.where.not(lessons: {result: nil})`
to
`@categories = current_user.categories.where.not(lessons: {result: nil})`